caluma/data_source/data_source_handlers.py

```diff
@@ -0,0 +1,68 @@
+from collections import namedtuple
+from datetime import timedelta
+
+from caluma.data_source.utils import cache_handler
+from caluma.extensions import data_sources
+
+DataSource = namedtuple("DataSource", ["name", "info"])
+
+
+class DataSourceException(Exception):
+    pass
+
+
+def is_iterable_and_no_string(value):
+    try:
+        iter(value)
+        if isinstance(value, str):
+            return False
+        return True
+    except TypeError:
+        return False
+
+
+class Data:
+    def __init__(self, data):
+        self.data = data
+        self.option, self.value = self.load()
+
+    def load(self):
+        if is_iterable_and_no_string(self.data):
+            if not 0 < len(self.data) < 3:
+                raise DataSourceException(f"Failed to parse data:\n{self.data}")
+            elif len(self.data) == 1:
+                return str(self.data[0]), str(self.data[0])
+            return str(self.data[0]), str(self.data[1])
+
+        return str(self.data), str(self.data)
+
+
+def get_data_sources():
+    sources = []
+    for name in dir(data_sources):
+        cls = getattr(data_sources, name)
+        if (
+            isinstance(cls, type)
+            and name.endswith("DataSource")
```

Comment:
Shall we still explicitly mention in the docs that the datasources have to end with "DataSource" in order to be picked up?

Reply:
Additionally we could also add a note on this to the "No data_source found" error message.

Reply:
Hmm I find this a rather strange convention that's not used in any other place in Django. Sure we need this? I mean requiring the name to end with a given suffix to be picked up. Could lead to surprising results and/or support cases

Reply:
Absolutely! I can't say why, but I implemented auto-discovery for data_sources.

This is fixed now.